### ISSUE TITLE
Cloning repo to desktop: ...mingw64\...\git-submodule: line 1121: cmd_: command not found

### ISSUE THREAD

**Issue by aubergine10** [desktop/desktop#9089]
### Describe the bug

Trying to clone a repo to desktop, I get this error:

```
C:\....\GitHubDesktop\app-2.3.1\resources\app\mingw64\libexec\git-core\git-submodule: line 1121: cmd_: command not found
```

### Version & OS

* GitHub Desktop Version 2.3.1
* Win ver: Windows 10 Home 64 Bit 1909 (build 18363.592)

### Steps to reproduce the behavior

1. Complete reinstall of GitHub Desktop
    * Remove old version via Add/Remove Programs
    * Delete GitHub Desktop folders from %appdata% - both Local and Roaming
    * Download installer from https://desktop.github.com/
    * Install
    * Reboot
2. First attempt at cloning
    * Try cloning https://github.com/krzychu124/Cities-Skylines-Traffic-Manager-President-Edition
    * Failed initially with issue mentioned in #3096
    * Added program specific exclusion for `.....git\usr\bin\sh.exe` to not do mandatory ASLR
    * That solved that particular problem
    * Note: Did **not** add any other exclusions (ie. the exploit settings txt file linked near end of #3096)
3. Second attempt
    * Delete part cloned repo from disk
    * Try cloning https://github.com/krzychu124/Cities-Skylines-Traffic-Manager-President-Edition
    * Resulted in error `command not found`

### Expected behavior

Repo should be cloned without error.

### Actual behavior

```
C:\....\GitHubDesktop\app-2.3.1\resources\app\mingw64\libexec\git-core\git-submodule: line 1121: cmd_: command not found
```

### Screenshots

![image](https://user-images.githubusercontent.com/1386719/74206009-8b482d00-4c71-11ea-939e-b13c6eb4f3ab.png)

### Logs

[2020-02-11.desktop.production.log](https://github.com/desktop/desktop/files/4183853/2020-02-11.desktop.production.log)

### Additional context

Not sure if it makes any difference, but I've set windows to use Windows PowerShell instead of normal command prompt. I can't remember how I ended up doing that; it was during a spate of hardware/driver fail and I was doing lots of DISM and SFC stuff, so probably in a guide about that. Currently both DISM and SFC are reporting no issues at all.

**Comment by niik** [comment]
Hey @aubergine10, thanks for opening this, and thank you especially for your detailed description and clear reproduction steps.

I was able to reproduce your issue by turning on mandatory ASLR and excluding `sh.exe` only. The offending line in the submodule shell script uses `echo` and `sed` as well as `sh` but I didn't try adding these as exceptions. As described in #3096 Git for Windows isn't able to operate under the constraints of ASLR.

Unfortunately there's not much we can do to work around this as far as I understand. Without significant changes to msys certain parts of Git for Windows will not function with ASLR turned on. 

As such I'm afraid the only recommendation I can give is to either turn off mandatory ASLR or experiment with adding every executable under `\resources\app\git\` to the exception list as described in https://github.com/desktop/desktop/issues/3096#issuecomment-529138491. Note that the path to the git executables will change with every release of GitHub Desktop so I'd look for a way to specify the exceptions less strictly than by absolute path.

**Comment by aubergine10** [comment]
As a quick test I turned ASLR off completely, and can confirm that fixed the issue. 👍 

So yes, looks like I'm going to have to add a whole bunch of exclusions to the ASLR thing.

**Comment by maxzhu273** [comment]
I know this issue is three years old but since I found this issue by google, thought I'd put some more info for those also stumbling upon this from google. Keep in mind that I ran into the error message in the title of this issue while running CMake for O3DE, so I wouldn't know whether Github Desktop needs a third exe to be excluded from mandatory ASLR.

I disabled mandatory ASLR for `Git/usr/bin/sed.exe` and that seems to make the `cmd_: command not found` error go away. In total, I have mandatory ASLR disabled for `sed.exe` and `sh.exe`.

For future reference (in case the list of exes that need ASLR disabled changes), I found `sed.exe` by looking at what processes were being spawned by git/cmake using sysinternals Process Explorer.
